core/frontend/src/store/popups.ts

```diff
@@ -0,0 +1,67 @@
+import Vue from 'vue'
+import {
+  Action, getModule, Module, Mutation, VuexModule,
+} from 'vuex-module-decorators'
+
+import store from '@/store'
+import { ActivePopup, PopupOptions, PopupResult } from '@/types/popups'
+
+@Module({
+  dynamic: true,
+  store,
+  name: 'popups',
+})
+class PopupsStore extends VuexModule {
+  popups: Record<string, ActivePopup> = {}
+
+  @Mutation
+  addPopup(popup: ActivePopup): void {
+    Vue.set(this.popups, popup.id, popup)
+  }
+
+  @Mutation
+  removePopup(id: string): void {
+    Vue.delete(this.popups, id)
+  }
+
+  @Mutation
+  close(id: string): void {
+    if (this.popups[id]) {
+      this.popups[id].dismissed = true
+      this.popups = { ...this.popups }
+    }
+  }
+
+  @Mutation
+  closeAll(): void {
+    Object.keys(this.popups).forEach((id) => {
+      this.popups[id].dismissed = true
+    })
```

Comment:
Yeah, usually I prefer the Object.keys to avoid needing to use the hasOwn check because of the possible inherited keys that the for in maybe can include, but in this case as it is a simple record and I already tested its not happening, so the suggestion is already added!